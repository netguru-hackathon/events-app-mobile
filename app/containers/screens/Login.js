import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';
import {
  Dimensions,
  Image,
  Text,
  TouchableHighlight,
  View,
} from 'react-native';
import Config from 'react-native-config';
import colors from '../../constants/colors';
import { SlackOAuthUrl } from '../../constants/constants';
import permissionScopes from '../../utils/oauth';
import signInWithSlack from '../../img/SignInWithSlack.png';
import I18n from '../../utils/translations';

const { width } = Dimensions.get('window');

const Container = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
`;

const LoginButton = styled.Image`
  width: ${0.7 * width};
`;

const authURL = `${SlackOAuthUrl}?client_id=${Config.SLACK_CLIENT_ID}&scope=${permissionScopes()}`;

const Login = props =>
  <Container>
    <Text>{I18n.t('welcome')}</Text>
    <TouchableHighlight
      underlayColor={colors.TRANSPARENT}
      onPress={() => props.navigation.navigate('SlackAuth', { authURL })}
    >
      <LoginButton
        source={signInWithSlack}
        resizeMode={'contain'}
      />
    </TouchableHighlight>
  </Container>;

Login.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};

export default Login;
