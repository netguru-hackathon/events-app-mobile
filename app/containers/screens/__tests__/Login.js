import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../Login';

test('renders correctly', () => {
  const tree = renderer.create(
    <Login
      navigation={{
        navigate: () => {},
      }}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
