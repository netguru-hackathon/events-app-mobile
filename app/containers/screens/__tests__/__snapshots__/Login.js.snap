// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Welcome
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "width": 525,
        }
      }
    />
  </View>
</View>
`;
